huaweicloud/services/ecs/resource_huaweicloud_compute_instance.go

```diff
@@ -311,11 +311,10 @@ func ResourceComputeInstance() *schema.Resource {
 				Set: resourceComputeSchedulerHintsHash,
 			},
 			"user_data": {
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-				// just stash the hash for state & diff comparisons
-				StateFunc: utils.HashAndHexEncode,
+				Type:             schema.TypeString,
+				Optional:         true,
+				StateFunc:        utils.HashAndHexEncode,
+				DiffSuppressFunc: utils.SuppressUserData,
```

Comment:
please support a comment to tell others what this function does without entry the function.

Reply:
If got base64 encrypted data or user_ data is in plaintext, it will not be modified.